ckanext/validation/plugin/__init__.py

```diff
@@ -153,12 +155,19 @@ def _process_schema_fields(self, data_dict):
 
         return data_dict
 
-    def before_create(self, context, data_dict):
+    if ckan_2_10:
```

Comment:
This condition shouldn't really be needed. The function can exist on 2.9, and be called when `is_dataset` is False (instead of duplicating its effects).

So you could have something like:

    def before_resource_create(..):
        <do stuff>

    def before_create(...):
        if not self._data_dict_is_dataset(data_dict):
            return self.before_resource_create(context, data_dict)

Reply:
@ThrawnCA can You please check this [PR](https://github.com/frictionlessdata/ckanext-validation/pull/98) if this issue is implemented OK?